src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -257,58 +254,53 @@ internal static ProcessInfo[] GetProcessInfos(Predicate<int>? processIdFilter =
             ProcessInfo[] processInfos;
 
             // Start with the default buffer size.
-            int bufferSize = DefaultCachedBufferSize;
+            int bufferSize = DefaultCachedBufferSize * sizeof(long);
 
             // Get the cached buffer.
-            long[]? buffer = Interlocked.Exchange(ref CachedBuffer, null);
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(bufferSize);
```

Comment:
Would it make sense to use a dedicated buffer pool instance to mitigate against unrelated components misusing the shared instance? Not sure what our best practices are here.

Reply:
The best practice here is to use `Shared` and avoid having multiple dedicated pools (as this grows the number of allocated arrays and defeats the purpose)